cli/src/commands/codegen.rs

```diff
@@ -107,6 +107,14 @@ fn codegen(
         type_substitutes,
         crate_path,
     );
-    println!("{runtime_api}");
+    match runtime_api {
+        Ok(runtime_api) => println!("{runtime_api}"),
+        Err(e) => {
+            // `Span` cannot be sent between threads.
```

Comment:
I don't understand this comment, is that the reason why you can't propagate the error?

Reply:
The `CodegenError` includes a `Span` in some variants that can be used to hint at better compile errors.
The `color_eyre` does impose `StdError + Send + Sync` on the returned error, to circumvent that I've just printed the error message in red. I'll update the comment :D